Doc/library/sqlite3.rst

```diff
@@ -616,18 +635,27 @@ Connection objects
    .. method:: commit()
 
       Commit any pending transaction to the database.
-      If there is no open transaction, this method is a no-op.
+      If :attr:`autocommit` is ``True``, or there is no open transaction,
+      this method does nothing.
+      If :attr:`!autocommit` is ``False``, a new transaction is implicitly
+      opened if a pending transaction was committed by this method.
```

Comment:
CAM's wording flows better IMO.

Reply:
I actually like the active voice introduced by this change as its more direct and (to me) a bit clearer to follow, but I also find the "if committing succeeded" wording to be rather awkward and potentially confusing. As an alternative, you _could_ consider

```suggestion
      If :attr:`!autocommit` is ``False``, this method implicitly opens a new transaction
      if it committed a pending transaction.
```

Reply:
Yeah, that's better than what's proposed, but I'm not sure it is worth changing it.

Reply:
Yeah, I had made it right before you merged it, but only submitted it after. If @maggyero or you make a followup PR for further docs refinements, it can be addressed there.

Reply:
The first time I read the current version of this sentence I honestly did not understand it. Only after reviewing the implementation of `commit()` (and `rollback()`) I understood that the ‘if a pending transaction was committed by this method’ refers to the same `commit()` call (i.e. `commit()` first tries to commit then begin):

```c
else if (self->autocommit == AUTOCOMMIT_DISABLED) {
         if (connection_exec_stmt(self, "COMMIT") < 0) {
             return NULL;
         }
         if (connection_exec_stmt(self, "BEGIN") < 0) {
             return NULL;
         }
     }
```